### PR TITLE
[v4][Banner] Remove withContext

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

Part of #1484

### WHAT is this pull request doing?

I'm converting Banner to a functional component and removing `withContext`

## 

### How to 🎩
Tests / percy / development server

### CODE REVIEW

File: src/components/Banner/Banner.tsx [hunk 1]
```diff
@@ -47,135 +37,127 @@ export interface Props {
...
+export default function Banner({
+  icon,
+  action,
+  secondaryAction,
+  title,
+  children,
+  status,
+  onDismiss,
+}: Props) {
+  const {withinContentContainer} = React.useContext(WithinContentContext);
```
Comment (AndrewMusgrave) [inline src/components/Banner/Banner.tsx]: A lot of red for not many changes. It's mostly class -> function with the removal of HoC and this addition

File: src/components/Banner/Banner.tsx [hunk 2]
```diff
@@ -206,10 +188,3 @@ function secondaryActionFrom(action: Action) {
     </button>
   );
 }
-
-export default compose<Props>(
-  withContext<Props, WithAppProviderProps, WithinContentContextType>(
-    WithinContentContext.Consumer,
-  ),
-  withAppProvider(),
```
Comment (AndrewMusgrave) [inline src/components/Banner/Banner.tsx]: This is never being used

File: src/components/Banner/Banner.tsx (hunk 1 above)
Comment (BPScott) [reply src/components/Banner/Banner.tsx]: Diff settings -> "hide whitespace changes" shall be our new best friend for a lot of these convert to function versions

Comment (AndrewMusgrave) [reply src/components/Banner/Banner.tsx]: Definitely 🙏 , luckily we don't have many components to transform 😄

**Comment by AndrewMusgrave** [comment]
This got fixed when master was merged into v4

**Pull request closed without merging**